### ISSUE TITLE
Access to run number, proposal number in DataCollection

### ISSUE THREAD

**Issue by leguyader** [European-XFEL/EXtra-data#168]
Once a DataCollection is created via open_run, it is not easy to figure out which proposal and run number the DataCollection is based on, also if raw or proc data were used.

It could be handy to have this accessible. For example something like:
```
run = ed.open_run(proposal, runNB)
print(run.get_run_number())
```

**Comment by takluyver** [comment]
This could relate to #27, because the run number & proposal number are now stored in the files.

It's a bit more complicated than it might look in EXtra-data because the DataCollection object can span more than one run (e.g. by using the [union()](https://extra-data.readthedocs.io/en/latest/reading_files.html#extra_data.DataCollection.union) method). So we need to work out if it should return the value from an arbitrary file, or validate that there's just one value before returning it, or have an option to return a set of values, or...

**Comment by leguyader** [comment]
Yes, union make things not so simple. I can imagine cases where data are recorded in different proposal by mistake (i.e. data recorded still in a commissioning proposal from the week before before switching to recording data in the dedicated user proposal).

I think the simplest is that we track which runs and proposals are used by other means.

Thank you for your input.

**Comment by tmichela** [comment]
Do we have a feel of how often runs are merged, and if that is useful to have these properties in such case?
In [this project](https://git.xfel.eu/gitlab/tmichela/EXtra-HED) which is a thin wrapper around extra-data, I do expose `run` and `proposal` numbers, but set them to `None` if DataCollections are merged.
That is the easy solution, but that may solve the majority of cases? We could also return a collection of (proposal, run) sets?

**Comment by takluyver** [comment]
My guess is that it's not very common. But to do anything better than picking an arbitrary value requires it to be somewhat aware of runs as a grouping, because it currently just sees a pool of files.
